pkg/virtual/workspaces/cmd/cmd.go

```diff
@@ -117,11 +131,40 @@ func (o *WorkspacesAPIServer) RunAPIServer(stopCh <-chan struct{}) error {
 		return err
 	}
 	kcpInformer := kcpinformer.NewSharedInformerFactory(kcpClient, 10 * time.Minute)
+	
+	utilruntime.Must(tenancyv1alpha1.AddToScheme(legacyscheme.Scheme))
+	legacyscheme.Scheme.SetVersionPriority(tenancyv1alpha1.SchemeGroupVersion)
+	
 	rootAPIServerBuilder := virtualrootapiserver.RootAPIServerBuilder {
-		AdditionalInformers: []internalinterfaces.SharedInformerFactory {
-			kcpInformer.(internalinterfaces.SharedInformerFactory),
-			
+		InformerStarts: []func(stopCh <-chan struct{}) {
+			kcpInformer.Start,			
+		},
+		RootPathresolver: func(urlPath string, requestContext context.Context) (accepted bool, prefixToStrip string, completedContext context.Context) {
+			completedContext = requestContext
+			if path := urlPath; strings.HasPrefix(path, "/services/applications/") {
+				path = strings.TrimPrefix(path, "/services/applications/")
+				i := strings.Index(path, "/")
+				if i == -1 {
+					return 
+				}
+				workspacesScope := path[:i]
+				if workspacesScope != "personal" && workspacesScope != "organization" && workspacesScope != "global" {
+					return
+				}
+				return true, "/services/applications/" + workspacesScope + "/", context.WithValue(requestContext, "VirtualWorkspaceWorkspacesScope", workspacesScope)
```

Comment:
Do we have a specific reason for `"/services/applications/"`? What else do we do other than services? Are there services that are not applications?

Reply:
Please find some more context about this in the following thread: https://kubernetes.slack.com/archives/C021U8WSAFK/p1637259445021500?thread_ts=1637169437.446200&cid=C021U8WSAFK